openapi/sled-agent.json

```diff
@@ -872,75 +915,38 @@
         "maxLength": 63
       },
       "NetworkInterface": {
-        "description": "A `NetworkInterface` represents a virtual network interface device.",
+        "description": "Information required to construct virtual network interfaces for a guest",
         "type": "object",
         "properties": {
-          "description": {
-            "description": "human-readable free-form text about a resource",
-            "type": "string"
-          },
-          "id": {
-            "description": "unique, immutable, system-controlled identifier for each resource",
-            "type": "string",
-            "format": "uuid"
-          },
-          "instance_id": {
-            "description": "The Instance to which the interface belongs.",
-            "type": "string",
-            "format": "uuid"
-          },
           "ip": {
-            "description": "The IP address assigned to this interface.",
             "type": "string",
             "format": "ip"
           },
           "mac": {
-            "description": "The MAC address assigned to this interface.",
-            "allOf": [
-              {
-                "$ref": "#/components/schemas/MacAddr"
-              }
-            ]
+            "$ref": "#/components/schemas/MacAddr"
           },
           "name": {
-            "description": "unique, mutable, user-controlled identifier for each resource",
-            "allOf": [
-              {
-                "$ref": "#/components/schemas/Name"
-              }
-            ]
-          },
-          "subnet_id": {
-            "description": "The subnet to which the interface belongs.",
-            "type": "string",
-            "format": "uuid"
+            "$ref": "#/components/schemas/Name"
           },
-          "time_created": {
-            "description": "timestamp when this resource was created",
-            "type": "string",
-            "format": "date-time"
+          "slot": {
+            "type": "integer",
+            "format": "uint8",
+            "minimum": 0
           },
-          "time_modified": {
-            "description": "timestamp when this resource was last modified",
-            "type": "string",
-            "format": "date-time"
+          "subnet": {
+            "$ref": "#/components/schemas/IpNet"
```

Comment:
I think you're pointing out that a single guest interface will eventually allowed to have multiple IP addresses, potentially of different versions and in different subnets. If that's right, then yes, this will need to change to support that. My understanding of the V1 goal was that we're achieving that functionality in the guest by supporting multiple NICs. RFD 21 sec 10.2 explicitly labels multiple IPs per NIC as a future direction, thankfully!

Re-reading your comment though, I'm not sure that's what you meant. Can you elaborate a bit?

Reply:
The RFD 21 10.2 bit was around multiple IPv4 addresses or multiple IPv6 addresses on a single interface. It was meant to allow both a single IPv4 and IPv6 address on one interface (one from each version of the VPC subnet there).

Reply:
Ah, I misunderstood that then. I'm going to defer that work a bit, since the public API currently only allows exactly one address of either protocol. I'll create an issue to track that.

Reply:
Certainly doesn't have to be part of this and something we can re-evaluate in terms of scope somewhere else. Just wanted to help clarify intent.

Reply:
Filed https://github.com/oxidecomputer/omicron/issues/958 to track this.